packages/js/src/ai-optimizer/components/ai-optimize-button.js

```diff
@@ -70,28 +72,41 @@ const AIOptimizeButton = ( { id, isPremium = false } ) => {
 	// (4) all blocks are in visual mode.
 	// eslint-disable-next-line complexity
 	const { isEnabled, ariaLabel } = useSelect( ( select ) => {
-		if ( activeAIButtonId !== null && ! isButtonPressed ) {
-			return {
-				isEnabled: false,
-				ariaLabel: null,
-			};
-		}
+		const keyphraseAssessments = [ "introductionKeyword", "keyphraseDensity", "keyphraseDistribution" ];
+		if ( keyphraseAssessments.includes( id ) ) {
+			// Create a Paper object with current content to use consistent validation
+			const paper = new Paper( content || "", { keyword: keyword || "" } );
+			const hasValidKeyphrase = paper.hasKeyword();
+			// const hasValidContent = paper.hasText();
+
+			// Check global disabled reasons first (for unsupported content)
+			const disabledAIButtons = select( "yoast-seo/editor" ).getDisabledAIFixesButtons();
+			if ( Object.keys( disabledAIButtons ).includes( aiOptimizeId ) ) {
+				// Always show the global disabled reason when content is unsupported
+				return {
+					isEnabled: false,
+					ariaLabel: disabledAIButtons[ aiOptimizeId ],
+				};
+			}
 
-		const disabledAIButtons = select( "yoast-seo/editor" ).getDisabledAIFixesButtons();
-		if ( Object.keys( disabledAIButtons ).includes( aiOptimizeId ) ) {
-			return {
-				isEnabled: false,
-				ariaLabel: disabledAIButtons[ aiOptimizeId ],
-			};
+			// If no keyphrase is set, disable the button and show the relevant tooltip
+			if ( ! hasValidKeyphrase ) {
+				return {
+					isEnabled: false,
+					ariaLabel: __( "Please add both a keyphrase and some text to your content.", "wordpress-seo" ),
+				};
+			}
```

Comment:
With this approach, when there is no keyphrase and no text, the feedback is: “None of the elements in your content is supported by Yoast AI Optimize. This means you cannot use this feature with your current content.”
I am not sure if this approach makes sense.

For me, the following scenarios makes more sense:
1. No keyphrase and no text: “Please add both a keyphrase and some text.”
2. No keyphrase, but there is text: “Please add both a keyphrase and some text.”
3. There is keyphrase but no text: “Please add both a keyphrase and some text.”
4. There is keyphrase and text, but none of the text is supported: “None of the elements in your content is supported by Yoast AI Optimize. This means you cannot use this feature with your current content.”

Reply:
To check for scenario no. 4, in https://github.com/Yoast/wordpress-seo-premium/blob/a29ef814b0b5ae33bb40db3751d11cfe1cd24922/assets/js/src/ai/ai-fix-assessments/helpers/disable-ai-buttons.js#L16, we only want to return “None of the elements in your content is supported by Yoast AI Optimize. This means you cannot use this feature with your current content.” if:
1. the token's length is 0, and
2. the text from the Paper `const collectData = get( window, "YoastSEO.analysis.collectData", false );` is not empty